homeassistant/components/alarm_control_panel/ifttt.py

```diff
@@ -0,0 +1,106 @@
+"""
+Interfaces with alarm control panels that have to be controlled through IFTTT.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/alarm_control_panel.ifttt_securitysystem/
+"""
+import logging
+import voluptuous as vol
+
+import homeassistant.components.alarm_control_panel as alarm
+from homeassistant.components.alarm_control_panel import PLATFORM_SCHEMA
+from homeassistant.components.ifttt import (
+    ATTR_EVENT, DOMAIN as IFTTT_DOMAIN, SERVICE_TRIGGER)
+from homeassistant.const import STATE_ALARM_DISARMED, CONF_NAME, CONF_CODE
+import homeassistant.helpers.config_validation as cv
+
+DEPENDENCIES = ['ifttt']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "Home"
+
+EVENT_ALARM_ARM_AWAY = "alarm_arm_away"
+EVENT_ALARM_ARM_HOME = "alarm_arm_home"
+EVENT_ALARM_ARM_NIGHT = "alarm_arm_night"
+EVENT_ALARM_DISARM = "alarm_disarm"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_CODE): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up a control panel managed through IFTTT."""
+    name = config.get(CONF_NAME)
+    code = config.get(CONF_CODE)
+
+    alarmpanel = IFTTTAlarmPanel(hass, name, code)
+    add_devices([alarmpanel])
+
+
+class IFTTTAlarmPanel(alarm.AlarmControlPanel):
+    """Representation of an alarm control panel controlled throught IFTTT."""
+
+    def __init__(self, hass, name, code):
```

Comment:
Don't pass in `hass`. It will be set on the entity when it has been added to home assistant.